server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/QuotaChangesMethod.scala

```diff
@@ -0,0 +1,102 @@
+
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_QUOTA, JMAP_CORE)
+
+  val quotaChangesResolver: QuotaChangesResolver = QuotaChangesResolver(quotaManager, quotaRootResolver)
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier], invocation: InvocationWithContext, mailboxSession: MailboxSession, request: QuotaChangesRequest): Publisher[InvocationWithContext] =
+    quotaChangesResolver.getLatestState(mailboxSession.getUser, capabilities)
+      .map(newState => QuotaChangesResponse.from(request.sinceState, UuidState.fromJava(newState), request.accountId))
+      .map(response => InvocationWithContext(
+        invocation = Invocation(
+          methodName = methodName,
+          arguments = Arguments(QuotaSerializer.serializeChanges(response)),
+          methodCallId = invocation.invocation.methodCallId),
+        processingContext = invocation.processingContext))
+      .onErrorResume {
+        case e: CanNotCalculateChangesException => SMono.just(InvocationWithContext(Invocation.error(ErrorCode.CannotCalculateChanges, e.getMessage, invocation.invocation.methodCallId), invocation.processingContext))
+        case e => SMono.error(e)
+      }
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): Either[Exception, QuotaChangesRequest] =
+    QuotaSerializer.deserializeQuotaChangesRequest(invocation.arguments.value) match {
+      case JsSuccess(mailboxGetRequest, _) => Right(mailboxGetRequest)
+      case errors: JsError => Left(new IllegalArgumentException(ResponseSerializer.serialize(errors).toString))
+    }
+
+}
+
+case class QuotaChangesResolver(private var quotaManager: QuotaManager,
+                                private var quotaRootResolver: UserQuotaRootResolver) {
+
+  private val jmapQuotaManagerWrapper: JmapQuotaManagerWrapper = JmapQuotaManagerWrapper(quotaManager, quotaRootResolver)
+
+  def getLatestState(username: Username, capabilities: Set[CapabilityIdentifier]): SMono[State] =
+    jmapQuotaManagerWrapper.retrieveQuotaRoot(username, capabilities)
+      .flatMap(quotaRoot => SMono(quotaManager.getQuotasReactive(quotaRoot)))
+      .collectSeq()
+      .map(quotasList => evaluateStateFromQuotas(username, capabilities, quotasList))
+
+
+  private def evaluateStateFromQuotas(username: Username, capabilities: Set[CapabilityIdentifier], quotasList: Seq[QuotaManager.Quotas]): State = {
+    if (!quotasList.exists(quotas => quotas.getMessageQuota.getLimit.isLimited || quotas.getStorageQuota.getLimit.isLimited)) {
+      throw new CanNotCalculateChangesException("Quota is empty, hence we cannot calculate changes")
+    }
+
+    val usedTotal: Long = quotasList
+      .map(quotas => quotas.getMessageQuota.getUsed.asLong() + quotas.getStorageQuota.getUsed.asLong())
+      .sum
+    State.of(UUID.nameUUIDFromBytes(s"${username.toString}:${capabilities.map(_.value).mkString("_")}:${usedTotal}".getBytes(StandardCharsets.UTF_8)))
+  }
```

Comment:
Should we calculate the change for the limit also? E.g a user limit changed from unlimited to limited and the opposite?

Reply:
`unlimited to limited` in JmapQuota context I think it is from no quota -> has a quota.

Reply:
but right now unlimited quota is not returned in Quota/get.
If we change Quota/Get for it, what is the value of "limit"? It is an `UnsignedInt` type, and the real value from the server is `0`.

Reply:
We agreed on unlimited quota <-> no quota -> no need to return it in Quota/get. However, we can just plus calculate the unlimited quota in the hash/state formula?

But I am ok with limit addition in the hash formula now, maybe calculating also the unlimited case is over complex for this naive impl.